jabbercat/models.py

```diff
@@ -1,28 +1,27 @@
 import bisect
-import collections.abc
 import enum
 import typing
 import unicodedata
-
-import lxml.builder
-import lxml.etree
+from itertools import chain
+from typing import List
 
 import aioxmpp.callbacks
 import aioxmpp.structs
-
+import jabbercat.avatar
 import jclib.conversation
 import jclib.identity
 import jclib.instrumentable_list
 import jclib.metadata
 import jclib.roster
 import jclib.utils
+import lxml.builder
+import lxml.etree
+from aioxmpp.callbacks import Signal
+from jclib.roster import AbstractRosterItem
```

Comment:
Please group the imports properly. The order should be, separated by empty lines:
1. Python standard library
2. Other libraries (each library with its own block)
3. aioxmpp
4. jclib
5. jabbercat

Within each block, ``import`` statements should be before ``from ... import ...``. 

This is inspired by and compatible with the respective statement in [PEP 8 #Imports](https://www.python.org/dev/peps/pep-0008/#imports).

Reply:
This is the result of applying pycharm's "optimize imports" (ctrl+shift+I IIRC)

For me it's a good-enough trade-off because it gives stable and correct imports (in fact, the tool spotted needless imports along the way), but I understand the force of habit and that's why I made it into its own changeset. I'll roll it back.